uat/driver/test_bundle.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+from pathlib import Path
+
+import pytest
+from lightkube import ApiError, Client, codecs
+from lightkube.generic_resource import create_global_resource, load_in_cluster_generic_resources
+from utils import assert_namespace_active, delete_job, fetch_job_logs, wait_for_job
+
+log = logging.getLogger(__name__)
+
+ASSETS_DIR = Path("assets")
+JOB_TEMPLATE_FILE = ASSETS_DIR / "test-job.yaml.j2"
+PROFILE_TEMPLATE_FILE = ASSETS_DIR / "test-profile.yaml.j2"
+
+TESTS_DIR = os.path.abspath(Path("integrations"))
+TESTS_IMAGE = "kubeflownotebookswg/jupyter-scipy:v1.7.0"
```

Comment:
This could probably be a simpler image (I don't think you need all the kubeflow notebook machinery, more just something with jupyter?).  But having said that, I don't think it adds value to simplify unless we ever run into problems.

Reply:
Regarding [this comment](https://github.com/canonical/bundle-kubeflow/pull/649/files#r1277712362), having this IMAGE specified here opens us to version incompatibilities.  Probably it wont matter (likely last version's scipy image has what we need for this version's tests), but still.  That alone might be a good argument for going with a very basic image here (python, or a generic scipy image) and installing all the requirements we need so the tests always come with everything.

Reply:
I wanted to ensure the most straightforward path between testing/developing the example notebooks from inside a Notebook server to actually running these as part of the UATs. That said, I haven't looked into the other images provided, perhaps a different one could provide the same advantages with a leaner set of dependencies.